src/types/agent.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from "zod";
+
+// TODO: Fix role type declaration
+const promptMessageSchema = z.object({
+  role: z.enum(["system", "user", "assistant"]),
+  //   characterName: z.enum(["hal", "dot", "ash", "eve", "roy", "host"]),
```

Comment:
Is this useful?

Reply:
I want to use it in the next PR for injecting the right name in the prompt.